lib/tapioca/dsl/helpers/graphql_type_helper.rb

```diff
@@ -78,6 +79,13 @@ def type_for(argument)
 
         sig { params(constant: Module).returns(String) }
         def type_for_constant(constant)
+          if constant.instance_methods.include?(:prepare)
+            prepare_method = constant.instance_method(:prepare)
+            if (prepare_signature = Runtime::Reflection.signature_of(prepare_method))
+              return prepare_signature.return_type&.to_s if prepare_signature.return_type
```

Comment:
I think this could fall into a similar situation as https://github.com/Shopify/tapioca/pull/1659#discussion_r1383597406 with void/NotTyped. Can we account for those and add tests for them by changing the signature of `prepare`?

Reply:
I added tests (small refactor to the `GraphqlTypeHelper` to make it reusable), the thing to note is that if the `prepare` method has no signature, I'm assuming it's the `InputObject#prepare` method from the graphql-ruby gem which returns the Input type, and not `Untyped`